volunteer_base/models/volunteer_engagement.py

```diff
@@ -0,0 +1,191 @@
+        required=True,
+        tracking=True,
+        states={'finished': [('readonly', True)]},
+        track_visibility='always'
+    )
+    state = fields.Selection(
+        string='State',
+        selection = [
+            ('new', 'New'),
+            ('active', 'Active'),
+            ('onhold', 'On Hold'),
+            ('success', 'Sucessful'),
+            ('cancel', 'Cancelled')],
+        default='new',
+        required=True,
+        tracking=True,
+        track_visibility='always',
+        help=" * The 'New' status is the default when an engagement is created.\n"
+             " * The 'Active' status is used when the volunteer is actively working in the engagement.\n"
+             " * The 'On Hold' status is used when the volunteer is pausing his activities in an engagemen.\n"
+             " * The 'Successful' status is set when the volunteer has successfully ended to engagement.\n"
+             " * The 'Cancelled' status is used when a volunteer has aborted an engagnement befor successfully ending it."
+    )
+    start_date = fields.Date(
+        string='Start date',
+        tracking=True,
+        states={'finished': [('readonly', True)]},
+        track_visibility='always',
+        help="In this field the start date of an engagement can be saved."
+    )
+    end_date = fields.Date(
+        string='End date',
+        tracking=True,
+        states={'finished': [('readonly', True)]},
+        track_visibility='always',
+        help="In this field the end date of an engagement can be saved."
+    )
+    state_onhold_until = fields.Date(
+        string='On hold until',
+        tracking=True,
+        track_visibility='always',
+        help="In this field that date when a volunteering engagmeent was set on hold"
+    )
+    date_cancel = fields.Date(
+        string='Cancel date',
+        help="Date when Volunteer requested to cancel the engagement"
+    )
+    cancel_reason_id = fields.Many2one(
+        string='Cancel reason',
+        comodel_name='engagement.cancel.reason',
+        index=True,
+        track_visibility='onchange',
+        help="The reason why a volunteer caneceled an engagement"
+    )
+    onhold_reason_id = fields.Many2one(
+        string='On hold reason',
+        comodel_name='engagement.onhold.reason',
+        index=True,
+        track_visibility='onchange',
+        help="The reason why a volunteer puts an engagement on hold"
+    )
+
+    filerable_groupable_fields = ['state','state_onhold_until','start_date','end_date']
+
+    _sql_constraints = [
+        ('volunteer_engagement_unique_name', 'UNIQUE (name)',
+         'The Engagement Reference must be unique!'),
+    ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('name', '/') == '/':
+                vals['name'] = self.env['ir.sequence'].next_by_code('volunteer.engagement')
+        return super(VolunteerEngagement, self).create(vals_list)
```

Comment:
With @api.multi you have to expect, that self contains multiple objects.
Loop through self and apply your code to the single object where the loop is pointing to.

I would also suggest to create the super object first and then apply your object manipulations.